docs/rest_api.rst

```diff
@@ -663,3 +663,139 @@ be rejected.
   :statuscode 404: If the channel does not exist
   :statuscode 409: If the given block number argument is invalid
   :statuscode 500: Internal Raiden node error
+
+
+.. http:get:: /api/1/payments
+
+   Query for all channel history events (EventPaymentSentSuccess, EventPaymentSentFailed, EventPaymentReceivedSuccess) on all the channels.
+
+   **Example Request**:
+
+   .. http:example:: curl wget httpie python-requests
+
+      GET /payments HTTP/1.1
+      Host: localhost:5001
+
+  **Example Response**:
+
+  .. sourcecode:: http
+
+     HTTP/1.1 200 OK
+     Content-Type: application/json
+
+    [
+        {
+            block_number: 3694255,
+            event: "EventPaymentReceivedSuccess",
+            amount: 5,
+            initiator: "0x82641569b2062B545431cF6D7F0A418582865ba7"
+        },
+        {
+            block_number: 3694245,
+            event: "EventPaymentSentSuccess",
+            amount: 35,
+            target: "0x82641569b2062B545431cF6D7F0A418582865ba7"
+        },
+        {
+            block_number: 3694216,
+            event: "EventPaymentSentSuccess",
+            amount: 20,
+            target: "0x82641569b2062B545431cF6D7F0A418582865ba7"
+        }, ...
+    ]
+
+  :statuscode 200: For successful query
+  :statuscode 400: If the provided query string is malformed
+  :statuscode 404: If the channel does not exist
+  :statuscode 500: Internal Raiden node error
+
+
+.. http:get:: /api/1/payments/(token_address)
+
+   Query for all channel history events (EventPaymentSentSuccess, EventPaymentSentFailed, EventPaymentReceivedSuccess) tied to a specific token network.
+
+   **Example Request**:
+
+   .. http:example:: curl wget httpie python-requests
+
+      GET /payments/0x0f114A1E9Db192502E7856309cc899952b3db1ED HTTP/1.1
+      Host: localhost:5001
+
+  **Example Response**:
+
+  .. sourcecode:: http
+
+     HTTP/1.1 200 OK
+     Content-Type: application/json
+
+    [
+        {
+            block_number: 3694255,
+            event: "EventPaymentReceivedSuccess",
+            amount: 5,
+            initiator: "0x82641569b2062B545431cF6D7F0A418582865ba7"
+        },
+        {
+            block_number: 3694245,
+            event: "EventPaymentSentSuccess",
+            amount: 35,
+            target: "0x82641569b2062B545431cF6D7F0A418582865ba7"
+        },
+        {
+            block_number: 3694216,
+            event: "EventPaymentSentSuccess",
+            amount: 20,
+            target: "0x82641569b2062B545431cF6D7F0A418582865ba7"
+        },...
+    ]
+
+  :statuscode 200: For successful query
+  :statuscode 400: If the provided query string is malformed
+  :statuscode 404: If the token_address does not exist
+  :statuscode 409: If the token_address argument is invalid
+  :statuscode 500: Internal Raiden node error
+
+.. http:get:: /api/1/payments/(token_address)/(target_address)
```

Comment:
I think you we're right the first time. Let's just make one point for the endpoint and say that token and target are optional. Sorry for that.